### PR TITLE
Fix regressions in #4651

### PR DESCRIPTION
**Pull request by crusaderky**
Fix regressions introduced in #4651:
- failing test
- if a worker sends a heartbeat before its SystemMonitor has a chance to run, metrics["memory"] will be None and that will cause a crash in the scheduler.

### CODE REVIEW

**Comment by pentschev** [comment]
I confirmed locally this fixes the problem for me too, thanks so much for the quick fix @crusaderky !

**Comment by crusaderky** [comment]
Stress test outcome:
- 76x successful
- 1x an unrelated test failed
- 22x CI fell over with no log whatsoever, very fast and definitely before reaching the test suite

File: distributed/scheduler.py
Comment (crusaderky) [inline distributed/scheduler.py:3843]: I can't see a way to write a unit test for this short of spoofing Client.hearbeat?

Comment (pentschev) [reply distributed/scheduler.py:3843]: It seems like it's going to be the only way, another test does exactly that: https://github.com/dask/distributed/blob/e4b534af5722a7aebb8d5b086081984e93e1784f/distributed/tests/test_worker.py#L1697-L1717

**Comment by pentschev** [comment]
Would people be ok with merging this even without a test and leaving a test for a different PR? This would unblock CI in Dask-CUDA and UCX-Py (potentially other RAPIDS projects too).

File: distributed/scheduler.py
Comment (fjetter) [reply distributed/scheduler.py:3843]: There is `distributed.admin.system-monitor.interval` which controls how often the monitor runs. You could set it to incredibly high values such that it is never executed during the test runtime

----

Another patch version w/ using monkeypatch, you could remove the PC before it is even started. Something like

```python
@pyters.mark.asyncio
async def test_foo():
    s = Scheduler()
    s.periodic_callbacks["monitor"] = None
    w = Worker(s)
    w.periodic_callbacks["monitor"] = None
    await s
    await w
    async with Client(s):
        ....

```

**Comment by fjetter** [comment]
I'm fine with postponing tests since our CI is also failing hard. @crusaderky your call

**Comment by crusaderky** [comment]
I need an extra 2-3 hours to cook a unit test. I'm ok merging without

File: distributed/tests/test_scheduler.py
Comment (jrbourbeau) [inline distributed/tests/test_scheduler.py:2388]: Is there something more direct we can probe here instead of sleeping for 10 seconds?

**Review by jrbourbeau — APPROVED** [review]
Thanks for your work here @crusaderky. I'm going to merge this as is to unblock CI, but left a comment below

File: distributed/tests/test_scheduler.py
Comment (crusaderky) [reply distributed/tests/test_scheduler.py:2388]: Not really because the unmanaged memory is very volatile, so we don't know how many keys are going to be spilled out exactly. Also, as noted it doesn't slow the test down.

File: distributed/scheduler.py
Comment (crusaderky) [reply distributed/scheduler.py:3843]: 1.
Setting ``distributed.admin.system-monitor.interval`` to a very high value, before I create the Scheduler has no effect.

2.
Setting 
```
s.periodic_callbacks["monitor"] = None
w.periodic_callbacks["monitor"] = None
```
fails with 
```
        for pc in self.periodic_callbacks.values():
>           pc.stop()
E           AttributeError: 'NoneType' object has no attribute 'stop'
```

3.
Setting
```
pc = PeriodicCallback(lamba: None, 999999999)
s.periodic_callbacks["monitor"] = pc
w.periodic_callbacks["monitor"] = pc
```
has no effect.

**Pull request merged**
